WalletWasabi/Nito/AsyncEx/AsyncMutex.cs

```diff
@@ -401,18 +401,29 @@ public static async Task WaitForAllMutexToCloseAsync()
 				}
 			}
 
+			var lastLog = DateTime.MinValue;
+			var lastNumberOfAliveMutexes = 0;
+
 			while (true)
 			{
 				lock (AsyncMutexesLock)
 				{
-					bool stillRunning = AsyncMutexes.Any(am => am.Value.IsAlive);
-					if (!stillRunning)
+					int numberOfAliveMutexes = AsyncMutexes.Count(am => am.Value.IsAlive);
```

Comment:
It could be useful to keep the alive mutexes in a variable, in that way you can use it below too. Otherwise, in really rare cases your could log "Waiting for:" with an empty list.

```suggestion
					var aliveMutexes = AsyncMutexes.Where(am => am.Value.IsAlive);
					int numberOfAliveMutexes = aliveMutexes.Count();
```